samples/bluetooth/hci_rpmsg/src/main.c

```diff
@@ -218,11 +224,14 @@ static int hci_rpmsg_send(struct net_buf *buf)
 	net_buf_push_u8(buf, pkt_indicator);
 
 	LOG_HEXDUMP_DBG(buf->data, buf->len, "Final HCI buffer:");
-	ipc_service_send(&hci_ept, buf->data, buf->len);
+
+	do {
+		err = ipc_service_send(&hci_ept, buf->data, buf->len);
+	} while (err == RPMSG_ERR_NO_BUFF);
```

Comment:
Yes I think it would spin without yielding. During debugging, I just added a `k_msleep(10)` when failing to send, but removed it before the submission.
I did not think this was a problem (except maybe for logging ?) since by default main thread has a prio of 0, which is pre-emptable by coop threads.
Would you suggest having a `k_yield` call when the error code is != 0 ? That'd be ok with me.

I'll also add a warning after a reasonable timeout: the openamp docs are currently wrong (well I think it's the timing calculation that is) and the timeout is not 15 seconds, but rather 150ms. So I think not a warning every failed allocation, that'd be too much noise, but a warning every second would be okay. Do you agree ?